pkg/controller/migmigration/backup.go

```diff
@@ -358,6 +443,93 @@ func (t *Task) annotateStorageResources() error {
 		}
 	}
 
+	return nil, resticPodCount
+}
+
+func (t *Task) areStagePodsCreated() (bool, error) {
+	client, err := t.getSourceClient()
+	if err != nil {
+		return false, err
+	}
+	// check if we have already created dummy pods
+	// if so, return true
+	uniqueBackupLabelKey := fmt.Sprintf("%s-%s", pvBackupLabelKey, t.PlanResources.MigPlan.UID)
+	labelSelector := map[string]string{
+		fmt.Sprintf("%s-copy", uniqueBackupLabelKey): "true",
+	}
+	options := k8sclient.MatchingLabels(labelSelector)
+	podList := corev1.PodList{}
+	err = client.List(context.TODO(), options, &podList)
+	if err != nil {
+		return false, err
+	}
+	podCount := len(podList.Items)
+	if podCount == 0 {
+		return false, nil
+	}
+	readyCount := 0
+	for _, pod := range podList.Items {
+		if pod.Status.Phase == corev1.PodRunning {
+			readyCount += 1
+		}
+	}
+	// Check if all pods are ready
+	if readyCount == podCount {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (t *Task) createStagePods() error {
+	client, err := t.getSourceClient()
+	if err != nil {
+		return err
+	}
+
+	// Dummy pods haven't been created yet, lets create them
+	uniqueBackupLabelKey := fmt.Sprintf("%s-%s", pvBackupLabelKey, t.PlanResources.MigPlan.UID)
+	labelSelector := map[string]string{
+		uniqueBackupLabelKey: pvBackupLabelValue,
+	}
+	options := k8sclient.MatchingLabels(labelSelector)
+	podList := corev1.PodList{}
+	err = client.List(context.TODO(), options, &podList)
+	if err != nil {
+		return err
+	}
+	for _, pod := range podList.Items {
+		if pod.Annotations[resticPvBackupAnnotationKey] == "" {
+			continue
+		}
+		dummyPod := corev1.Pod{}
+		dummyPod.Spec.Containers = pod.Spec.Containers
+		dummyPod.Spec.Volumes = pod.Spec.Volumes
+		// Swap image ref, command, and args and create a new dummy pod to be backed up
+		for i, _ := range dummyPod.Spec.Containers {
+			dummyPod.Spec.Containers[i].Image = rhel7ImageRef
+			dummyPod.Spec.Containers[i].Command = []string{"sleep"}
+			dummyPod.Spec.Containers[i].Args = []string{"infinity"}
+		}
+		dummyPod.Name = fmt.Sprintf("%s-dummy-pod", pod.Name)
+		dummyPod.Namespace = pod.Namespace
+		if dummyPod.Labels == nil {
+			dummyPod.Labels = make(map[string]string)
+		}
+		dummyPod.Labels[uniqueBackupLabelKey] = "true"
+		dummyPod.Labels[fmt.Sprintf("%s-copy", uniqueBackupLabelKey)] = "true"
+		if dummyPod.Annotations == nil {
+			dummyPod.Annotations = make(map[string]string)
+		}
+		dummyPod.Annotations[resticPvBackupAnnotationKey] = pod.Annotations[resticPvBackupAnnotationKey]
+		err = client.Create(context.TODO(), &dummyPod)
+		// This shouldn't happen with check at beginning of function, but we don't
```

Comment:
@dymurray could you elaborate on what check happened at the beginning of the function that will prevent an error from occurring here? And why don't we want to return an error?

Reply:
Sorry, referenced old code. With the `areStagePodsCreated` being implemented I removed the check at the beginning of the function to prevent duplication. Changed the comment.